doc/concepts/index.rst

```diff
@@ -17,11 +15,43 @@ To be able to simulate a model, three basic steps need to be considered:
 2. Define the computational resources available to execute the model;
 3. Initiate and execute a simulation of the recipe on the chosen hardware resources.
 
-The python front-end further abstracts away some of these steps for single cell models, where users only need to
+
+.. graphviz::
```

Comment:
I'm unsure about this graph:
1. It shows some of the methods of a recipe, but not all.
2. It doesn't show any info about what goes into a context or a domain decomposition. Why is the recipe special? 
3. The nodes of the graph are identical, which indicates that their contents are similar types. But the nodes that make up a recipe aren't objects, they are method definitions. Maybe they can have a different color to indicate that?

Reply:
1. You are right. I'm seeking a compromise between showing everything, and then ending up with a figure that's no longer helpful for getting an overview. I also didn't want to add ellipsis nodes (seemed like more clutter). The caption 'Outline' was meant to convey this is not a complete reference.
2. The recipe is special because it contains the elements of what most neuroscientists expect to see in an overview over a morphologically detailed simulator: where are the cells and the network? The recipe has so far been the structure they are most interested in/confused by when I speak them, so I think it's a good idea to tell them up-front that that's where the biology goes; what they usually describe as a model.
    'context' should be 'execution context', which might make it clearer what it is about. However, 'domain decomposition' is perhaps a bit too cryptic. I want to both match what we actually call these concepts later on though to not add to the new-nomenclature overload. What would you think is a good solution? Maybe we could circle 'recipe' and it's ingredients and annotate it with 'neuroscience' or so, and 'execution context+'domain decomp.' with 'computational' or so?
3. Cell descriptions such as `cable_cell` are objects too no? The 'cell kind' and 'number of cells' not, you're right. The description below could have a word or two on cell desc+kinds. 'number of cells' is self-descriptive enough I'd say.

Reply:
Removed for now.